### PR TITLE
Copter: move zigzag exit into mode_zigzag.cpp

### PR DESCRIPTION
**Pull request by tatsuy**
moving zigzag exit function from Copter::set_mode to ModeZigZag::exit()

### CODE REVIEW

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -363,6 +356,12 @@ void Copter::exit_mode(Mode *&old_flightmode,
     }
 #endif
 
+#if MODE_ZIGZAG_ENABLED == ENABLED && SPRAYER_ENABLED == ENABLED
```
Comment (peterbarker) [inline ArduCopter/mode.cpp]: This shouldn't be conditional upon the sprayer being enabled.

File: ArduCopter/mode_zigzag.cpp [hunk 2]
```diff
@@ -41,6 +41,15 @@ bool ModeZigZag::init(bool ignore_checks)
     return true;
 }
 
+// perform cleanup required when leaving zigzag mode
+void ModeZigZag::exit()
+{
+    // The pump will stop if the flight mode is changed from ZigZag to other
+    if (g2.zigzag_auto_pump_enabled) {
```
Comment (peterbarker) [inline ArduCopter/mode_zigzag.cpp]: This needs pragma protection against SPRAYER_ENABLED being false, doesn't it?

File: ArduCopter/mode.cpp (hunk 1 above)
Comment (tatsuy) [reply ArduCopter/mode.cpp]: Thank you for the review.
Fixed it.

File: ArduCopter/mode_zigzag.cpp (hunk 2 above)
Comment (tatsuy) [reply ArduCopter/mode_zigzag.cpp]: Fixed it, thank you.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**